unic/ucd/normal/src/canonical_combining_class.rs

```diff
@@ -12,44 +12,98 @@
 
 //! Accessor for Canonical_Combining_Class (ccc) property
 
-// TODO: [API] Support Canonical_Combining_Class_Values, either by creating a separate `enum` type,
-// or converting `CanonicalCombiningClass` into enum and allowing access to numerical value.
-// <http://unicode.org/reports/tr44/#Canonical_Combining_Class_Values>
-
-// TODO: Use associate type `u8`
 
 use std::cmp::Ordering;
 
 
 /// Represents *Canonical_Combining_Class* property of a Unicode character.
 ///
 /// * <http://unicode.org/reports/tr44/#Canonical_Combining_Class>
-pub trait CanonicalCombiningClass {
-    /// If the *ccc* has value `Not_Reordered` (`0`).
-    fn ccc_is_not_reordered(&self) -> bool;
+#[derive(Copy, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct CanonicalCombiningClass(u8);
+
+
+// TODO: Once we fully adopt 1.20 change these to associated consts on CanonicalCombiningClass
+/// Canonical Combining Classes by their name
+#[allow(non_upper_case_globals)]
+pub mod values {
```

Comment:
I used `values` instead of `long_names` as, as noted here, I think of this case as associated values to be used _as if_ this were a `#[repr(u8)]` enum with all values, whether named or not, valid. The enums don't use a `long_names` module, just their self namespace. In any case, I'd expect `mod long_names` to allow me to use the `Long_Name` form, rather than the `RustName` form.

As for `abbr_names`, I don't really think it's needed here. Just use `use CanonicalCombiningClass as CCC` and `CCC(10)`; it's only two more characters and might even be slightly better in matching situations, depending on how Rust treats consts in a match. (Rust requires a `Eq` implementation to use a const in a pattern context which makes me think it's a `Eq` call instead of whatever magic match over regular patterns does.)